llvm/lib/Transforms/InstCombine/InstCombineAddSub.cpp

```diff
@@ -2631,6 +2631,18 @@ Instruction *InstCombinerImpl::visitSub(BinaryOperator &I) {
                                                /* IsNUW */ false))
       return replaceInstUsesWith(I, Res);
 
+  if (match(Op0, m_ZExt(m_PtrToInt(m_Value(LHSOp)))) &&
+      match(Op1, m_PtrToInt(m_Value(RHSOp))) && isa<GlobalValue>(RHSOp)) {
+    Value *Offset;
+    if (match(LHSOp, m_GEP(m_Specific(RHSOp), m_Value(Offset)))) {
```

Comment:
This is going to restrict to a single-offset GEP, which is unnecessary. You should directly start with a `dyn_cast<GEPOperator>` and then check getPointerOperand().